architecture-decision-records/0021-service-unavailable-page.md

```diff
@@ -0,0 +1,109 @@
+# 20. Deploying a service unavailable page 
+
+**Date:** 2021-07-14
+
+## Context
+We need a mechanism to manually deploy a service unavailable page. There are two main reasons we might want to do this. 
+* There is a serious issue with the site that we can't fix immediately.
+* We want to carry out maintenance which will affect the service.
+
+This will be a rare occurrence as almost all deployments will be done with zero downtime. 
+
+The page will be a single static html page which will be displayed when a user goes to any of the https://tdr.nationalarchives.gov.uk/* urls.
+
+![Service Unavailable Design](images/service-unavailable-design.png)
+
+## Evaluation criteria
+* We must be able to deploy the page manually.
+* We should avoid creating any permanently running services for this page as it will be used rarely.
+* The switch to the new page should only take seconds.
+* It should be easy for any developer on the TDR team to deploy.
+* It should be easy to switch back to the TDR app.
+
+## Options considered
+
+### Options for rendering the page
+![Page rendering options](images/failover-page-options.png)
+### Static site in S3 private bucket
+#### Advantages
+* Cheap
+* Easy to maintain. 
+* There is no web server, only static html/css
+
+#### Disadvantages
+* To use Cloudfront with custom DNS, we need to have an ACM certificate in the us-east-1 region.
+
+### Static site in S3 public bucket
+#### Advantages
+* Cheap
+* Easy to maintain. There is no web server, only static html/css
+#### Disadvantages
+* The bucket is public
+
+### AWS Lambda rendering static HTML
+
+This can either be through API gateway or through an existing load balancer depending on the option we choose for the failover.
+#### Advantages
+* Cheap
+* Can keep everything in eu-west-2
+#### Disadvantages
+* This has to be done using a web server of some kind, most likely in Python or Javascript. I've created an example project
```

Comment:
Maybe "We've prototyped this to check it would work" rather than "I've created an example project" for the ADR?